library/x509_crt.c

```diff
@@ -2951,23 +2951,138 @@ static int x509_crt_verify_chain(
     }
 }
 
+static int parse_ipv4( const char* h, char *addr )
+{
+    unsigned char ip[4];
+    int i;
+    const char *strt = h;
+    char *endp;
+    unsigned long v;
+    for( i = 0; i < 4; i++ )
+    {
+        v = strtoul( strt, &endp, 10 );
+        if( endp == strt || v > 255 )
+        {
+            return ( 0 );
+        }
+
+        if( i < 3 && *endp != '.' )
+        {
+            return ( 0 );
+        }
+        ip[i] = v & (unsigned char)0xFF;
+        strt = endp + 1;
+    }
+
+    memcpy( addr, ip, sizeof ip);
```

Comment:
Style: Please use parenthesis when using `sizeof`, here and any other place you use `sizeof` without parenthesis. See https://tls.mbed.org/kb/development/mbedtls-coding-standards for details